pkg/config/imports.go

```
@@ -0,0 +1,406 @@
+	atmosFilePathsAbsolute, err := convertToAbsolutePaths(atmosFilePaths)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert paths to absolute paths: %w", err)
+	}
+	// Prioritize and sort files
+	atmosFilePathsAbsolute = detectPriorityFiles(atmosFilePathsAbsolute)
+	atmosFilePathsAbsolute = sortFilesByDepth(atmosFilePathsAbsolute)
+	return atmosFilePathsAbsolute, nil
+}
+
+// Helper function to generate search patterns for extension yaml,yml
+func generatePatterns(path string) ([]string, error) {
```

golangci-lint (1.62.2)

generatePatterns - result 1 (error) is always nil

(unparam)